gtsam/linear/NoiseModel.h

```diff
@@ -752,6 +754,12 @@ namespace gtsam {
         double weight(double error) const {
           return 1.0 / (1.0 + std::abs(error) / c_);
         }
+        double residual(double error) const {
+          double absError = std::abs(error);
```

Comment:
Probably compiler will figure it out, but add const to all these?

Reply:
Yup added to these as well as function definitions for other M-estimators.